src/routes/embedded/account/backup-wallet/backup-wallet-qrcode.tsx

```diff
@@ -0,0 +1,102 @@
+import type { JWKInterface } from "@dha-team/arbundles";
+import { useState, useCallback, useEffect } from "react";
+import { toast } from "react-toastify";
+import { Flex } from "~components/common/Flex";
+import { CopyToClipboard } from "~components/CopyToClipboard";
+import { Text } from "~components/embed/ui";
+import { OnboardingCard } from "~components/embed/ui/molecules/card/onboarding-card/OnboardingCard";
+import { useEmbedded } from "~utils/embedded/embedded.hooks";
+import { useLocation } from "~wallets/router/router.utils";
+import type { LocalWallet } from "~wallets/wallets.types";
+import browser from "webextension-polyfill";
+import { dataToFrames } from "qrloop";
+import { freeDecryptedWallet } from "~wallets/encryption";
+import { QRCodeWrapper } from "~components/QRCodeWrapper";
+import { QRCodeLoop } from "~components/QRCodeLoop";
+import { useTheme } from "~components/embed/contexts/ThemeContext";
+
+export function AccountBackupWalletQrCodeEmbeddedView() {
+  const { isDarkMode } = useTheme();
+  const { navigate } = useLocation();
+  const { getDecryptedWallet } = useEmbedded();
+  const [isLoading, setIsLoading] = useState(false);
+  const [copied, setCopied] = useState(false);
+  const [frames, setFrames] = useState<string[]>([]);
+  const [decryptedWallet, setDecryptedWallet] = useState<LocalWallet<JWKInterface> | null>(null);
+
+  const fetchDecryptedWallet = useCallback(async () => {
+    try {
+      if (decryptedWallet) return;
+      setIsLoading(true);
+      const wallet = await getDecryptedWallet();
+      setDecryptedWallet(wallet);
+    } catch (error) {
+      toast.error("Error getting wallet");
+    } finally {
+      setIsLoading(false);
+    }
+  }, [getDecryptedWallet, decryptedWallet]);
+
+  useEffect(() => {
+    fetchDecryptedWallet();
+  }, [fetchDecryptedWallet]);
+
+  useEffect(() => {
+    if (decryptedWallet?.keyfile) {
+      setFrames(dataToFrames(JSON.stringify(decryptedWallet.keyfile)));
+      freeDecryptedWallet(decryptedWallet.keyfile);
+    }
+  }, [decryptedWallet?.keyfile]);
+
+  useEffect(() => {
+    return () => setFrames([]);
```

Comment:
This doesn't remove the old values from memory, as you are just creating a new array/reference, and the other one will still be around until the GC takes care of it. I guess the idea here was to implement something similar to `freeDecryptedWallet()`, so I think using an object would be better. That way, what we would be doing is keeping the same object/reference but mutating its props/values.

Still, I have many doubts about this given 1) that the same issue with new references being created apply when changing the props/values and 2) we are now hooked into React, so updating the object that's also inside the state might cause some issues or not work as expected, but let's try.